src/mmap_unix.rs

```diff
@@ -160,6 +176,47 @@ impl MmapRegion {
             file_offset,
             prot,
             flags,
+            owned: true,
+        })
+    }
+
+    /// Creates a MmapRegion instance for an externally managed mapping.
+    ///
+    /// # Arguments
+    /// * `addr` - Pointer to the start of the mapping. Must be page-aligned.
+    /// * `size` - The size of the memory region in bytes. Must be a multiple of the page size.
+    /// * `prot` - The desired memory protection of the mapping.
+    /// * `flags` - This argument determines whether updates to the mapping are visible to other
+    ///             processes mapping the same region, and whether updates are carried through to
+    ///             the underlying file.
+    ///
+    /// # Safety
+    ///
+    /// To use this safely, the caller must guarantee that `addr` and `size` define a region within
+    /// a valid mapping that is already present in the process.
+    pub unsafe fn build_raw(addr: *mut u8, size: usize, prot: i32, flags: i32) -> Result<Self> {
```

Comment:
Can you also add an `Example` section? It will help people find where to get defines for `prot` and `flags` from.
The example can be something like:
```
let addr = 0;
let size = unsafe { libc::sysconf(libc::_SC_PAGESIZE) as usize };
let prot = libc::PROT_READ | libc::PROT_WRITE;
let flags = libc::MAP_NORESERVE | libc::MAP_PRIVATE;
let r = unsafe { MmapRegion::build_raw(addr as *mut u8, size, prot, flags).unwrap() };
```

Reply:
Hm... actually, I'd expect `build_raw` to be used only in situations where the user doesn't have direct control of the mapping, which implies that `prot` and `flags` should correspond to the actual properties of the already existing mapping.

On a second look, I see the method's documentation for `prot` and `flags` is not clear about this. What do you think if we clarify that instead of adding an example?

Reply:
Sounds good.